onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -290,37 +356,19 @@ def destroy(self, request, *args, **kwargs):
         self.get_object().delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
 
-    def _get_tableau_headers(self, xform):
-        """
-        Return a list of headers for tableau.
-        """
-        def shorten(xpath):
-            xpath_list = xpath.split('/')
-            return '/'.join(xpath_list[2:])
-
-        header_list = [
-            shorten(xpath) for xpath in xform.xpaths(
-                repeat_iterations=1)]
-        header_list += [
-            ID, UUID, SUBMISSION_TIME, TAGS, NOTES, REVIEW_STATUS,
-            REVIEW_COMMENT, VERSION, DURATION, SUBMITTED_BY, TOTAL_MEDIA,
-            MEDIA_COUNT, MEDIA_ALL_RECEIVED
-        ]
-        return header_list
-
     @action(methods=['GET'], detail=True)
     def schema(self, request, **kwargs):
         self.object = self.get_object()
         if isinstance(self.object.content_object, XForm):
             xform = self.object.content_object
-            headers = self._get_tableau_headers(xform)
+            sub_with_largest_repeat = fetch_form_submissions(
```

Comment:
Could we have this function only occur once?

_I remember this being brought up before... but can't remember why we decided not to send repeat data flattened into one column?_

Reply:
If we were to send it the repeat data as csv value. Someone in tableau would still be able to easily split it and it would remove the need for us to figure out how many repeat columns we should send....

https://help.tableau.com/current/pro/desktop/en-us/split.htm#:~:text=To%20split%20a%20field%20automatically,-On%20the%20Data&text=In%20the%20Data%20pane%2C%20right,are%20created%20by%20the%20split.

Reply:
Hey @DavisRayM 
I have been able to investigate this option of packing all nested repeat data in CSVs.
By trying this, I mean, in our current implementation, I removed indexing values in the repeat data and packed them in a CSV value instead.
This also involved changes in how we generate the schema, to remove indexing in repeat columns. And just have one column for all repeat iterations for that question.
So the repeat data we previously sent to tableau was this:
``` code
{
    "plot_2__subplot_4__splot_name": null,
    "plot_1__subplot_1__splot_pos": "1",
    "plot_2__subplot_2__photo2": null,
    "plot_3__subplot_2__photo2": null,
    "plot_2__plot_owner": "Troy",
    "plot_4__plot_owner": null,
    "plot_3__plot_owner": null,
    "plot_4__subplot_1__splot_pos": null,
    "plot_2__photo1": "n/a",
    "plot_4__subplot_1__splot_name": null,
    "plot_1__subplot_4__photo2": null,
    "plot_1__subplot_2__splot_name": null,
    "plot_2__subplot_2__splot_name": "Gacharage",
    "plot_1__sub_plots": 1,
    "plot_1__subplot_3__photo2": null,
    "plot_1__subplot_2__splot_pos": null,
    "plot_2__subplot_4__splot_pos": null,
    "plot_2__sub_plots": 1,
    "plot_2__subplot_4__photo2": null,
    "plot_4__plot_pos": null,
    "plot_3__subplot_1__splot_pos": null,
    "plot_2__plot_pos": "2",
    "plot_1__subplot_1__photo2": null,
    "plot_1__subplot_4__splot_pos": null,
    "plot_4__subplot_3__splot_name": null,
    "plot_2__subplot_count": "1",
    "plot_2__subplot_1__splot_pos": null,
    "plot_1__plot_owner": "Troy",
    "plot_4__subplot_count": null,
    "plot_1__plot_pos": "1",
    "plot_1__subplot_count": "1",
    "plot_2__subplot_2__splot_pos": "1",
    "plot_3__plot_name": null,
    "plot_2__plot_name": "Second Plot",
    "plot_2__subplot_3__photo2": null,
    "plot_3__subplot_1__photo2": null,l,
    "plot_1__plot_name": "Haven",
    "plot_1__subplot_1__splot_name": null,
    "plot_1__photo1": "n/a",
}
```
Now is this is the data being sent to tableau:
```code
{
   "plot_plot_owner": "Troy, Troy, null, null"
   "plot_photo": "n/a, n/a, null, null",
   "plot_plot_name": "Haven, Second Plot, null, null",
   "plot_plot_pos": "1, 2, null, null",
   "plot_sub_plots": "1, 1, null, null",
   "plot_subplot_photo": "null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null",
   "plot_subplot_splot_pos": "null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null",
}
```

While Tableau is able to receive this data, this brings up concerns, with how this change of the structure may impact data transformation for the support and data team. 
Sending data in this structure also removes metric data visualisation as there will never be integer values sent in repeat data.
So I wanted to run this through @muneneg @faith-mutua @B-Cheye @samwata